test/e2e/performanceprofile/functests/utils/sriov/sriov.go

```diff
@@ -1 +1,117 @@
 package sriov
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	sriovk8sv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	sriovv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	sriovcluster "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/cluster"
+	sriovnetwork "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/network"
+	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/cluster"
+	testlog "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/log"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodes"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// OperatorNamespace is the namespace where the SR-IOV Operator is installed
+	OperatorNamespace = "openshift-sriov-network-operator"
+)
+
+func CreatePolicyAndNetwork(ctx context.Context, sriovInfos *sriovcluster.EnabledNodes, namespace, networkName, resourceName, metaPluginsConfig string) {
+	GinkgoHelper()
+	numVfs := 4
+	ppAndSRIOVNodes, err := getNodesWithPerformanceProfileAndSRIOV(sriovInfos.Nodes)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(len(ppAndSRIOVNodes)).To(Not(BeZero()))
+
+	node := ppAndSRIOVNodes[0]
+	sriovDevice, err := sriovInfos.FindOneSriovDevice(node)
+	Expect(err).ToNot(HaveOccurred())
+
+	testlog.Infof("creating SRI-OV devices for node=%q", node)
+	_, err = sriovnetwork.CreateSriovPolicy(testclient.SRIOVClient, "test-policy", OperatorNamespace, sriovDevice.Name, node, numVfs, resourceName, "netdevice")
+	Expect(err).ToNot(HaveOccurred())
+	waitStable()
+
+	Eventually(func() int64 {
+		testedNode, err := testclient.K8sClient.CoreV1().Nodes().Get(ctx, node, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		resNum, _ := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/"+resourceName)]
```

Comment:
ideally the discovery function (or one of its helper) should return the full resource name we need to consume. Here we add a component of guesswork